modules/runtime/nuxeo-common/src/main/java/org/nuxeo/common/xmap/registry/SingleRegistry.java

```diff
@@ -34,7 +34,7 @@
 
     protected Object contribution;
```

Comment:
`contribution` must be volatile too, as it's read by `getContribution()` which is potentially called from different threads.